### PR TITLE
Improved locking performance on .NET 9.0+

### PR DESCRIPTION
**Pull request by MarkCiliaVincenti**
The new System.Threading.Lock offers greater performance, as independent benchmarks show (eg https://steven-giesel.com/blogPost/4cf6d48a-ec9d-4c68-961c-31fd8d8c1340)

### CODE REVIEW

**Comment by rockfordlhotka** [comment]
Thank you @MarkCiliaVincenti

Do I have a contributor agreement from you?

**Comment by MarkCiliaVincenti** [comment]
No you don't, but I saw that the process is lengthy(ish) so I'd rather that you let me know whether or not you will consider this before I go ahead and send you actual signatures (and I don't even have a working printer at the moment).

**Comment by MarkCiliaVincenti** [comment]
> Thank you @MarkCiliaVincenti
> 
> Do I have a contributor agreement from you?

No you don't, but I saw that the process is lengthy(ish) so I'd rather that you let me know whether or not you will consider this before I go ahead and send you actual signatures (and I don't even have a working printer at the moment).

File: Source/Directory.Build.props [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <Project>
  <PropertyGroup>
-   <LangVersion>latest</LangVersion>
+   <LangVersion>preview</LangVersion>
```
Comment (rockfordlhotka) [inline Source/Directory.Build.props:3]: Note to @rockfordlhotka - add a work item to reset this to `latest` once .NET 9 is released.

**Review by rockfordlhotka — APPROVED** [review]

**Comment by rockfordlhotka** [comment]
@MarkCiliaVincenti the changes look fine to me, I appreciate the contribution. Please do send me a contributor agreement.

You can, perhaps, sign on a touch screen or something? Print the result to PDF and return it?

**Comment by MarkCiliaVincenti** [comment]
> @MarkCiliaVincenti the changes look fine to me, I appreciate the contribution. Please do send me a contributor agreement.
> 
> You can, perhaps, sign on a touch screen or something? Print the result to PDF and return it?

Brilliant. Let me see if I can get a 25 year old printer to work 😂

**Comment by MarkCiliaVincenti** [comment]
@rockfordlhotka email sent.

File: Source/Directory.Build.props (hunk 1 above)
Comment (MarkCiliaVincenti) [reply Source/Directory.Build.props:3]: @rockfordlhotka fixed in https://github.com/MarimerLLC/csla/pull/4302

**Pull request merged**